sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1763,6 +1763,21 @@ class Analyzer(override val catalogManager: CatalogManager)
     def expandStarExpression(expr: Expression, child: LogicalPlan): Expression = {
       expr.transformUp {
         case f1: UnresolvedFunction if containsStar(f1.arguments) =>
+          // SPECIAL CASE: We want to block count(table.*) because in spark, count(table.*) will
+          // be expanded while count(*) will be converted to count(1). They will produce different
+          // results and confuse users if there is any null values. For count(t1.*, t2.*), it is
```

Comment:
Why `count(t1.*, t2.*)` is well-defined but `count(table.*)` is not? Isn't `count(t1.*, t2.*)` also expended like `count(table.*)` too?

Reply:
because expand `count(t1.*, t2.*)` is not ambiguous, I think no one will argue that `count(t1.*, t2.*)` should equal `count(1)`. Also this usage is not supported by the common database (MySQL, oracle, pgsql), so we are not in conflict with others.
but `count(table.*)` is not same. spark SQL expands the columns, pgsql converts it to `count(1)`, MySQL, Oracle (as well as ANSI) thinks this should be disallowed.
So I think block `count(table.*)` can avoid ambiguous while keeping `count(t1.*, t2.*)` can lead less side effects.

Reply:
How about the case `select count(t.*, t.*) from values (1, null) t(a, b)`? Btw, If the other databases don't support the syntax, `count(t1.*, t2.*)`, we still need to keep supporting it?

Reply:
Thanks @maropu  for reviewing.

`select count(t.*, t.*) from values (1, null) t(a, b)` will output 0.

I'm fine with blocking `count(t1.*, t2.*)` as well but since in spark, we are allowing other similar cases that other databases don't support (and not follow ANSI), it's not harmful to keep `count(t1.*, t2.*)` as one more case. After all, introducing unnecessary behavior change (blocking `count(t1.*, t2.*)`) doesn't benefit users.
The similar usages:
count(col_a, col_b) - count multiple columns is not supported by pgsql, oracel. MySQL only support with distinct
count(struct_col.*) - expand columns in struct data type is not supported by the mentioned databases.

Reply:
Let's minimize the change for now. We can open a new PR to discuss if we should forbid other cases.